London10 - Afsha Hossain - JS Core 1 - Week 2

Afsha10 opened the pull request:
<!--

The title for your pull request should be made in this format

CITY CLASS_NO - FIRST_NAME LAST_NAME - MODULE - WEEK_NO

For example,

London Class 7 - Chris Owen - HTML/CSS - Week 1

Please complete the details below this message

-->

**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

Afsha10 committed:
Completed Mandatory part

SeyoungJoo commented on mandatory/2-function-creation.js:
```diff
@@ -15,26 +21,60 @@ function isAcceptableUser(userAge, isLoggedIn) {}
+    return "You Can Register";
+  } else if (age >= 90) {
+    return "You Don't Need To Register";
+  } else {
```
The last statement that returns false is not required. recommend to delete the last else statement as the first 3 conditions already covers all the cases you can have.

```
function canRegister(age) {
  if (age <= 12) {
    return "You Are Too Young To Register";
  } else if (age > 12 && age < 90) {
    return "You Can Register";
  } else {
    return "You Don't Need To Register";
```

SeyoungJoo commented on mandatory/2-function-creation.js:
```diff
@@ -15,26 +21,60 @@ function isAcceptableUser(userAge, isLoggedIn) {}
+  } else if (totalPrice > 0 && totalPrice < 200) {
+    return 0.95 * totalPrice;
+  } else {
+    return 0
```
I think it should return 1

Afsha10 committed:
Update 1-factorial.js

seyyednavid commented on mandatory/2-function-creation.js:
```diff
@@ -15,26 +21,60 @@ function isAcceptableUser(userAge, isLoggedIn) {}
+    return price2;
+  } else price1;
+}
 
```
well done Afsha .

Afsha10 replied:
Thank you Navid!

moneyinthesky replied:
Agreed, good job 👍 
It's good that you're thinking about the case where `price1` is equal to `price2`.

moneyinthesky commented on mandatory/1-fix-functions.js:
```diff
@@ -10,20 +10,18 @@
 
 */
 
```
Both of these look good to me 👍

moneyinthesky commented on mandatory/2-function-creation.js:
```diff
@@ -4,7 +4,13 @@
     1. the user should be 18 or older
     2. the user must be logged in
 */
```
Nice 👍 
Can you think about how you can make this code even shorter? Do you need the `if` statement? Can you do it with only a `return` statement?

moneyinthesky commented on mandatory/2-function-creation.js:
```diff
@@ -15,26 +21,60 @@ function isAcceptableUser(userAge, isLoggedIn) {}
   */
-function printOddNumbers(limit) {}
+
+
```
Perfect 👍

moneyinthesky commented on mandatory/2-function-creation.js:
```diff
@@ -45,7 +85,12 @@ function canRegister(age) {}
   )
 */
 
```
Looks good 👍

moneyinthesky commented on extra/1-factorial.js:
```diff
@@ -10,6 +10,16 @@
 
```
Looks like there's a lot going on here 😄 
For problems like this, I would suggest we take a step back from the code - and think about some of the Problem Solving techniques we covered in the workshop in class. Maybe start by thinking about how you would normally do something like this (without a computer) and try drawing it out on a whiteboard or paper. This may give you some ideas about how to do this in code.

moneyinthesky replied:
Well done for attempting this though! It's a tough one!

Afsha10 replied:
I wrote pseudo code and watched your videos on function and looping and managed to do this on my own in less 30 minutes. Thank you, Barath!

moneyinthesky commented:
Good job on this coursework @Afsha10  👍

Afsha10 committed:
Completed the extra part

Completed the extra part on my own after writing the pseudocode and watching Barath's function and looping videos.
